edgelet/edgelet-docker/src/error.rs

```diff
@@ -19,23 +19,16 @@ pub struct Error {
     inner: Context<ErrorKind>,
 }
 
-fn get_message(
-    error: DockerApiError<serde_json::Value>,
-) -> ::std::result::Result<String, DockerApiError<serde_json::Value>> {
-    let DockerApiError { code, content } = error;
-
-    match content {
+pub(crate) fn get_message(err: &DockerApiError<serde_json::Value>) -> Option<&str> {
```

Comment:
What's the advantage to returning &str here rather than Option<String>?  I.e. do message.to_owned() here rather than in the calling function?

Reply:
`get_message` should not assume that the caller wants an owned string. That is the caller's choice. Indeed you can see that the caller in `runtime.rs` doesn't need an owned string.

Reply:
https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control

Reply:
Got it.  Consider this comment resolved.